### PR TITLE
Onchain identity id type initialized flag

### PR DESCRIPTION
**Pull request by OBrezhniev**

### CODE REVIEW

File: contracts/state/StateV2.sol [hunk 1]
```diff
@@ -69,7 +69,7 @@ contract StateV2 is Ownable2StepUpgradeable, IState {
     ) public initializer {
         verifier = verifierContractAddr;
         _defaultIdType = defaultIdType;
-        _transitionEnabledForOnchainIdentity = true;
+        _defaultIdTypeInitialized = true;
```
Comment (AndriianChestnykh) [inline contracts/state/StateV2.sol:72]: Just call the `setDefaultIdType()` here

Comment (OBrezhniev) [reply contracts/state/StateV2.sol:72]: We can't because it will fail. Msg.sender is not owner here.

Comment (AndriianChestnykh) [reply contracts/state/StateV2.sol:72]: True. But the DRY is still violated. Better move the same code to internal method or modifier

Comment (volodymyr-basiuk) [reply contracts/state/StateV2.sol:72]: We can have `internal` setter, but still have to provide `external` method to be able to upgrade contract / change `_defaultIdType` value outside of contract. This `external` will call `internal` setter. As for me, better to have one `exteranal` method and some logic in initialize (even if it's duplication) rather that have 2 setters (internal&external).
@AndriianChestnykh what do you think ?

Comment (AndriianChestnykh) [reply contracts/state/StateV2.sol:72]: Better to have a logic in one place. And two auth .... **wrappers**!!!.
Yes, I don't agree with your definition of setters. These are NOT setters (setter should be only one and that's the point) but just two wrappers, each of each implements different auth logic: one of them - `initializer`, which is called by ProxyAdmin contract and doing some init checks, another - onlyOwner modifier, checking for the `msg.sender == owner`). 
You may add 10 more auth wrappers in the future, which will be external methods and that will be fine in therms of design.
What tells you that design is good? The answer is - MAINTEINABILITY. Now both auth wrappers and the setter can evolve independently.

Comment (volodymyr-basiuk) [reply contracts/state/StateV2.sol:72]: Agreed, create a new PR with `internal` setter & 1 `external` wrapper (with onlyOwner modifier) - https://github.com/iden3/contracts/pull/156/files

**Pull request merged**